### DISCUSSION TITLE
Assign different detectors to specific cameras

### DISCUSSION THREAD

**Discussion by leccelecce** [blakeblackshear/frigate#4954] [Ideas]
Use case: I have some cameras for security purposes, some for presence detection, and some for wildlife.

I would like to use the default frigate model for the first purpose, potentially a custom model for the second, and another for the third.

From reading the docs, I can see the option on a CPU detector to specify a custom model, so I assume I can configure multiple detectors with different models. But the docs and config appear to suggest I can't then say cam1, cam2, cam3 uses cpu1, cam4 uses cpu2, cam5 uses gpu1

_When using multiple detectors they will run in dedicated processes, but pull from a common queue of detection requests from across all cameras._

Is this possible? I'm running 0.12-beta1

**Comment by leccelecce** [comment]
(It goes without saying I could run multiple Frigate instances, given your excellent code for the app itself runs pretty lightweight, but I'd love to keep everything in one if possible)

**Comment by NickM-27** [comment]
Not currently possible, definitely something that I think will be possible in the future
